guides/plugins/apps/payment.md

```diff
@@ -0,0 +1,266 @@
+      $response = [ 'status' => 'authorize' ];
+
+      // this returns a json encoded response with the `shopware-app-signature` in the header
+      return $this->sign($response, $content['source']['shopId']);
+}
+```
+
+Instead of a successful response, you can also provide a failed response with a message (which will be logged) like this:
+
+```php
+$response = [
+    'status' => 'fail',
+    'message' => 'The customer failed to pass the credit check.',
+];
+```
+
+{% hint style="warning" %}
+Keep in mind that just by providing a `message`, the payment will default to status `fail`.
+{% endhint %}
+
+## Asynchronous payments
+
+Asynchronous payments are more complicated and rely on interaction with the user and therefore a redirect to the payment provider.
+For example, this might be an integration with _PayPal_ or _Stripe_.
+The following process applies:
+
+Shopware sends the first `pay` POST request which is supposed to start the payment with the payment provider.
+All necessary data is provided: the `order`, `orderTransaction` and a `returnUrl`, where the user should be redirected to once the payment process with the payment provider has been finished.
+If everything is correct and the payment process is ready to start, the response to this request should be a `redirectUrl`, where the user is redirected to by Shopware.
+In case the payment can't be started (for example because of missing credentials for the shop), this can also return a `fail` status and / or a `message`.
+
+```php
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/async/pay", name="async.pay", methods={"POST"})
+ */
+public function startAsyncPayment(Request $request): JsonResponse
+{
+    // more on this in section "Validation"
+    $this->checkSignature($request);
+
+    $content = json_decode($request->getContent(), true);
+      
+    // you can identify the transaction later on with the orderTransactionId
+    $transactionId = $content['orderTransaction']['id'];
+
+    // implement your logic here based on the information provided in $content
+    // you should save the transactionId for later identification in the second request and in the user redirection
+    $response = [ 'redirectUrl' => sprintf('https://payment.app/user/go/here/%s/', $transactionId) ];
+
+    // this returns a json encoded response with the `shopware-app-signature` in the header
+    return $this->sign($response, $content['source']['shopId']);
+}
+```
+
+The second `finalize` POST request will be called once the user has been redirected that your app or your payment provider should back to the shop.
+This second request is only provided with the `orderTransaction` for identification purposes.
+The response `status` value determines the outcome of the payment, e.g.:
+* `cancel` if the user has aborted the payment at the payment provider's site
```

Comment:
Paragraphs should be separated by a blank line, otherwise the linting / presentation of Gitbook has unexpected behaviour. A small table might be easier to read than simple bullet points.

| Status | Description |
| --- | --- |
| `cancel` | User has aborted the payment on the provider's site |
| `fail` | Payment has failed (e.g. missing funds) |